docarray/typing/tensor/abstract_tensor.py

```diff
@@ -116,3 +117,8 @@ def __docarray_from_native__(cls: Type[T], value: Any) -> T:
     def get_comp_backend() -> Type[AbstractComputationalBackend]:
         """The computational backend compatible with this tensor type."""
         ...
+
+    @abc.abstractmethod
+    def n_dim(self) -> int:
```

Comment:
I fear that this name can clash with future backend. I would rather call it `__docarray_n_dim__`